sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -194,33 +201,209 @@ private void sendBody(HttpRequest httpRequest, Object progressReporter, HttpURLC
                 }
+                return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders);
+            } else {
+                AccessibleByteArrayOutputStream outputStream = getAccessibleByteArrayOutputStream(connection);
+                return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders,
+                    BinaryData.fromByteBuffer(outputStream.toByteBuffer()));
             }
+        } catch (IOException e) {
+            throw LOGGER.logThrowableAsError(new UncheckedIOException(e));
+        } finally {
+            connection.disconnect();
+        }
+    }
 
-            AccessibleByteArrayOutputStream outputStream = new AccessibleByteArrayOutputStream();
+    private void processTextEventStream(HttpRequest httpRequest, HttpURLConnection connection, ServerSentEventListener listener) {
+        RetrySSEResult retrySSEResult;
+        try (BufferedReader reader
+                 = new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
+            retrySSEResult = processBuffer(reader, listener);
+            if (retrySSEResult != null) {
+                retryExceptionForSSE(retrySSEResult, listener, httpRequest);
+            }
+        } catch (IOException e) {
+            throw LOGGER.logThrowableAsError(new UncheckedIOException(e));
+        }
+    }
 
-            try (InputStream errorStream = connection.getErrorStream();
-                InputStream inputStream = (errorStream == null) ? connection.getInputStream() : errorStream) {
-                byte[] buffer = new byte[8192];
-                int length;
-                while ((length = inputStream.read(buffer)) != -1) {
-                    outputStream.write(buffer, 0, length);
+    private boolean isTextEventStream(Headers responseHeaders) {
+        return responseHeaders.get(HeaderName.CONTENT_TYPE) != null &&
+        responseHeaders.get(HeaderName.CONTENT_TYPE).getValue().equals(ContentType.TEXT_EVENT_STREAM);
+    }
+
+    /**
+     * Processes the sse buffer and dispatches the event
+     *
+     * @param reader   The BufferedReader object
+     * @param listener The listener object attached with the httpRequest
+     */
+    private RetrySSEResult processBuffer(BufferedReader reader, ServerSentEventListener listener) {
+        StringBuilder collectedData = new StringBuilder();
+        ServerSentEvent event = null;
+        try {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                collectedData.append(line).append("\n");
+                if (isEndOfBlock(collectedData)) {
+                    event = processLines(collectedData.toString().split("\n"));
+                    if (!Objects.equals(event.getEvent(), DEFAULT_EVENT) || event.getData() != null) {
+                        listener.onEvent(event);
+                    }
+                    collectedData.setLength(0); // clear the collected data
```

Comment:
I think we'd be better off setting `collectedData` to a new StringBuilder instead of using `setLength(0)`